Cybersecurity_Tools/Behavior-Based Script Detector/utils/file_loader.py

```
@@ -0,0 +1,198 @@
+        except UnicodeDecodeError:
+            # Try with different encoding
+            with open(file_path, 'r', encoding='latin-1') as f:
+                file_content = f.read()
+        
+        # Parse AST
+        try:
+            tree = ast.parse(file_content)
+            atok = asttokens.ASTTokens(file_content, tree=tree)
+            return file_content, tree, atok
+        except SyntaxError as e:
+            raise SyntaxError(f"Syntax error in {file_path}: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)